sign/rpmgensig.cc

```diff
@@ -702,14 +702,20 @@ static int rpmSign(const char *rpm, int deleting, int flags)
     unloadImmutableRegion(&sigh, RPMTAG_HEADERSIGNATURES);
     origSigSize = headerSizeof(sigh, HEADER_MAGIC_YES);
 
-    if (flags & RPMSIGN_FLAG_IMA) {
-	if (includeFileSignatures(&sigh, &h))
-	    goto exit;
-    }
-
-    if (flags & RPMSIGN_FLAG_FSVERITY) {
-	if (includeVeritySignatures(fd, &sigh, &h))
-	    goto exit;
+    /* Add file signatures (if requested and not a source rpm) */
+    if (!headerIsSource(h)) {
+	if (flags & RPMSIGN_FLAG_IMA) {
+	    if (includeFileSignatures(&sigh, &h))
+		goto exit;
+	}
+	if (flags & RPMSIGN_FLAG_FSVERITY) {
+	    if (includeVeritySignatures(fd, &sigh, &h))
+		goto exit;
+	}
+    } else if (flags & (RPMSIGN_FLAG_IMA | RPMSIGN_FLAG_FSVERITY)) {
+	rpmlog(RPMLOG_WARNING,
```

Comment:
Maybe just make this an error? Then you could check this as the first as just error out without needing to touch the other logic.

Reply:
Indeed, my initial thought was "but you still want to sign the source rpm normally, even if file sigs aren't applied". But then, you should just drop `--signfiles` to achieve that. So yup, will fix.

Reply:
Hmm, even if we left this as a warning it'd make sense to test for source first, then you can warn and just disable the relevant flags, without needing to add more nested ifs.

Reply:
Yup, I even did that originally, I think, but then decided to do it this nested way (which is ugly, I completely agree), just don't remember *why* :sweat_smile: Probably not for any specific reason, though.

Will fix too, thanks!

Reply:
Yep, that was kinda my thinking here, too... If there's such a single command running somewhere regularly in the infra (which, as we know, uses all sorts of ways to sign packages, cough cough), it would start failing with this patch.

Reply:
Yup. And in that case, a warning suggests there's something bad in what they do, but is there? Realizing this use-case, I don't think there is. So if anything, it should be a RPMLOG_DEBUG item actually, and otherwise just ignore the file signatures for src.rpms.

Reply:
Interesting take, will ponder about it after the meeting :smile:

Reply:
Implemented all of above.